Working With APIs lesson: Added link to additional resource section (#27510)
TheOdinProject/curriculum: javascript/asynchronous_javascript_and_apis/working_with_apis.md

This section contains helpful links to related content. It isn't required, so consider it supplemental.

> - [This article by Ryan Thelin, briefly decribes the external, security, and automated benfits of using an API. It also informs and explains to the reader of the dissimilarity of release policie tiers, in addition to the different API protocols like, Representational State Transfer or also known as (REST) APIs, Remote Procedure Call (RPC), and Simple Object Access Protocol (SOAP). Furthermore, the author of this composition quickly articulates the differences between architectural API styles; for example, service-oriented architecture (SOA) and microservice architecture.](https://www.educative.io/blog/what-is-an-api)

Reviewer: ```suggestion
- [This article about APIs by Ryan Thelin](https://www.educative.io/blog/what-is-an-api) covers some extra topics like API release policies, API protocols and API architectural styles.
```
Thanks for the update. In my opinion the subscription could be summarised with my above suggestion for example. What do you think? If you agree, feel free to click `commit suggestion` or commit an alternative if you'd like

Reviewer: Also, we generally only add a short description about the link between the square brackets so the whole paragraph doesn't turn into a link. I edited that as well :)

Author: This is so much better. Mine was defiantly a bit lengthy. Thank you. 😊
